Production/plugins/TauTupleProducer.cc

```diff
@@ -96,6 +97,37 @@ namespace tau_analysis {
         static float full5x5_hcalDepth2OverEcalBc(const Elec* ele) { return ele->full5x5_hcalDepth2OverEcalBc(); }
     };
 
+    template<typename Gamma, int cmssw_version>
+    struct GetGammaVer;
+
+    //CMSSW 10 --> 12
+    //hcalOverEcal()   -->  hadronicOverEm()
+    //hcalOverEcalBc() -->  hadTowOverEm()
+    template<typename Gamma>
+    struct GetGammaVer<Gamma, 12> {
+        static float hcalDepth1OverEcal(const Gamma* photon) { return photon->hadronicOverEm(1); }
+        static float hcalDepth2OverEcal(const Gamma* photon) { return photon->hadronicOverEm(2); }
+        static float hcalDepth1OverEcalBc(const Gamma* photon) { return photon->hadTowOverEm(1); }
+        static float hcalDepth2OverEcalBc(const Gamma* photon) { return photon->hadTowOverEm(2); }
+        static float full5x5_hcalDepth1OverEcal(const Gamma* photon) { return photon->full5x5_hadronicOverEm(1); }
+        static float full5x5_hcalDepth2OverEcal(const Gamma* photon) { return photon->full5x5_hadronicOverEm(2); }
+        static float full5x5_hcalDepth1OverEcalBc(const Gamma* photon) { return photon->full5x5_hadTowOverEm(1); }
+        static float full5x5_hcalDepth2OverEcalBc(const Gamma* photon) { return photon->full5x5_hadTowOverEm(2); }
+    };
+
+    template<typename Gamma>
+    struct GetGammaVer<Gamma, 10> {
+        static float hcalDepth1OverEcal(const Gamma* photon) { return photon->hadronicDepth1OverEm(); }
+        static float hcalDepth2OverEcal(const Gamma* photon) { return photon->hadronicDepth2OverEm(); }
+        static float hcalDepth1OverEcalBc(const Gamma* photon) { return photon->hadTowDepth1OverEm(); }
+        static float hcalDepth2OverEcalBc(const Gamma* photon) { return photon->hadTowDepth2OverEm(); }
+        static float full5x5_hcalDepth1OverEcal(const Gamma* photon) { return -999.; }
+        static float full5x5_hcalDepth2OverEcal(const Gamma* photon) { return -999.; }
+        static float full5x5_hcalDepth1OverEcalBc(const Gamma* photon) { return -999.; }
+        static float full5x5_hcalDepth2OverEcalBc(const Gamma* photon) { return -999.; }
+    };
+
+
```

Comment:
@msessini please resolve the existing conflict in this part of the code.

And add overloading for version == 11 like we do for Electrons:
```c++
    template<typename Elec>
    struct GetElecVer<Elec, 11> : GetElecVer<Elec, 10> {};
```

Reply:
It would make sense in general, but I still use UL for displaced taus. Also: what is the release for phase 2, isn't it 11_*_*?

Reply:
@shedprog - for UL it is safe to move to 12_4_X for miniAOD->TauTuple step. Actually we need to do this to include DeepTau v2p5 in the next ntuples.
@DennRoy - is it ok to move to >= 12_X for the future Phase 2 studies?

Reply:
There are at least two changes required to make Phase2 production work with CMSSW 12. These are discussed here:
https://cms-talk.web.cern.ch/t/getting-muon-station-value-0/15765/1

Reply:
@DennRoy, does CMSSW_12_3_6 seem right for phase2 production as replacement of CMSSW_11_2_5 ?

Reply:
I'm not sure about any "best" release. I've used 12_4_0 and 12_4_8 before, which seems to have worked equally well. But I've also been told that 12_6_0_pre3 MIGHT fix the low tau reco efficiency in the Phase2 endcap (to be confirmed).